### ISSUE TITLE
Why does an <li role=button... fail... seems to read ok with screen reader

### ISSUE THREAD

**Issue by DavidMacDonald** [dequelabs/axe-core#1942]
Why does an <li role=button.... fail... seems to read ok with screen reader

**Comment by WilcoFiers** [comment]
That's a fair question. It's something I've been struggling with myself, and quite frankly, I don't have the answer yet. I'd appreciate any insight you might have.

Firstly, W3C specs clearly say this is invalid. The [ARIA spec](https://www.w3.org/TR/wai-aria-1.1/#list) says lists must contain either listitems or groups containing listitems. And [HTML in ARIA](https://www.w3.org/TR/html-aria/#el-li) says an `li` can not have `role=button`. That doesn't make it a WCAG failure though.

Going off memory, I thought that if you did do `li[role=button]` JAWS in IE will not count it as a list item. So that would be one problem.

Another concern would be with custom stylesheets. If your custom stylesheet is relying on all lists containing only listitems or groups of listitems, like the spec says they should, and for a button to be its own thing, then an `li[role=button]` could really throw off the presentation of that page with the custom stylesheet applied to it.


I'm not convinced this is a strong enough argument. Custom stylesheets don't seem to be as common as they once were. Even still, tools like Stylish are pretty darn useful. But to what extent someone might use it to change the presentation of a menu... I don't really know.

**Comment by WilcoFiers** [comment]
@DavidMacDonald, any thoughts?

**Comment by malouro** [comment]
Just to chime in a bit, I've ran into this while looking at [Material UI's collapsed Breadcrumbs example](https://material-ui.com/components/breadcrumbs/#collapsed-breadcrumbs), and realizing that it fails under an Axe scan due to this.

As a question for discussion - what would be an appropriate way to implement an expand button in lists with collapsed items?

Currently, the MUI Breadcrumbs look sorta like this:

```html
<nav aria-label="breadcrumbs">
	<ol>
		<li>
			<a href="/somewhere">First Breadcrumb</a>
		</li>
		<li aria-hidden="true"> / </li> <!-- separator -->

		<li role="button" aria-label="Show path" tabindex="0" aria-disabled="false" />
		<!-- Collapsed breadcrumbs not populated in the DOM at this point -->

		<li aria-hidden="true"> / </li> <!-- separator -->
		<li>
			<a href="/somewhere-else">Last Breadcrumb</a>
		</li>
	</ol>
</nav>
```

...where activating the button will expand the list to show all of the collapsed breadcrumbs.

Would it make more sense to make the button an actual `<button>` element and/or not worry about it being recognized as a `listitem`? Is `li[role=button]` an invalid choice for implementing this sort of pattern?

FWIW, I did some minor screen reading testing at the time and I recall NVDA not having an issue deciphering and reading off the number of list items, but can't recall how JAWS handled it. I can double-check and report back if it would help.

**Comment by eps1lon** [comment]
>Would it make more sense to make the button an actual <button> element and/or not worry about it being recognized as a listitem?

This would then violate HTML. 

Either way we break one spec or another. We're going to re-evaluate our pattern since it's the usual "design for sighted, bolt on a11y later" approach. It isn't clear to me how  a user with a screen reader would expand this list while navigating it anyway.

**Comment by WilcoFiers** [comment]
Haven't heard any good argument in favour of this pattern. This is definitely invalid ARIA, and there's a reasonable chance this can cause issues with custom stylesheets as well. Going to close the issue.

**Comment by gabycperezdias** [comment]
If I have a list of interactable items I use < ul > < li > shouldn't I be able to add a role `button` to the < li > otherwise how will SR know that this actually a button (not a link) is it _most correct_ to have a plain < li > with a button inside? Wouldn't this mess with Roving tabIndex to go over the items?
